diff --git a/node_modules/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt b/node_modules/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt
index ff4d53f..4ece231 100644
--- a/node_modules/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt
+++ b/node_modules/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt
@@ -340,6 +340,15 @@ class SQLiteModule : Module() {
     if (database.openOptions.enableChangeListener) {
       addUpdateHook(database)
     }
+    loadSQLiteVecExtension(database)
+  }
+
+  private fun loadSQLiteVecExtension(database: NativeDatabase) {
+    var errCode = database.ref.sqlite3_enable_load_extension(1)
+    if (errCode != NativeDatabaseBinding.SQLITE_OK) {
+        return
+    }
+    errCode = database.ref.sqlite3_load_extension("libsqlitevec", "sqlite3_vec_init")
   }
 
   @Throws(AccessClosedResourceException::class, SQLiteErrorException::class)
